test/aria/widgets/form/timefield/checkValue/TimeField.js

```diff
@@ -19,16 +19,115 @@ Aria.classDefinition({
     $constructor : function () {
         this.$TemplateTestCase.constructor.call(this);
         this.data = {
-            time1 : null,
-            time2 : null
-
+            time1 : null
         };
+        this.currentValue;
+        this.testValues = [];
+        this.checkValues = [];
+        this.separators = ["/", "\\", "h", ";", ",", ".", "-", ":"];
+        this.am = ["am", "a.m.", "a m"];
+        this.pm = ["pm", "p.m.", "p m"];
+        this.start();
         this.setTestEnv({
             data : this.data
         });
     },
     $prototype : {
+        start : function () {
+            this.check1Digit();
+            this.check2Digits();
+            this.check3Digits();
+            this.check4Digits();
+            this.checkAlphabetic();
+            this.checkValuesWithAMPM();
+        },
+        check1Digit : function () {
+            var hour, result, i;
+
+            // 1 digit = [h]
+            for (i = 0; i < 10; i++) {
+                hour = i + '';
+                result = '0' + i + ':00';
+                this.testValues.push(hour);
+                this.checkValues.push(result);
+            }
+        },
+        check2Digits : function () {
+            var hour, result, i;
+
+            // 2 digits = [h][h] (works with any of the separators after the first digit only)
+            for (i = 24; i > 22; i--) {
+                hour = i + '';
+                result = i !== 24 ? hour + ':00' : hour;
+                this.testValues.push(hour);
+                this.checkValues.push(result);
+            }
+            this.checkValuesWithSeparator(hour);
+        },
+        check3Digits : function () {
+            var hour, result;
+            // 3 digits = [h][h][m] (works with any of the separators after the first/second digit only)
+            hour = '150';
+            result = '15:00';
+            this.testValues.push(hour);
+            this.checkValues.push(result);
+            this.checkValuesWithSeparator(hour);
+        },
+        check4Digits : function () {
+            // 4 digits = [h][h]:[m][m] (works with any of the separators after the second digit only)
+            var value = "0000";
+            var result = "00:00";
+            this.testValues.push(value);
+            this.checkValues.push(result);
+            this.checkValuesWithSeparator(value);
+        },
+        checkAlphabetic : function () {
+            // alphabetic characters are invalid
+            this.testValues.push("12b", "b12", "1xx1", "xx");
+            this.checkValues.push("12b", "b12", "1xx1", "xx");
+
+        },
+        checkValuesWithAMPM : function () {
```

Comment:
I would create two distinct loops (despite the fact that in practice the two arrays have the same length).

Also I would rename some variables: 

- `items` :arrow_right: `index`
- `valid` :arrow_right: `baseInput` or something similar (you are building an also valid value with it)